.github/actions/akv-secret/index.js

```diff
@@ -0,0 +1,78 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+const fs = require('fs');
+
+(async () => {
+  try {
+    // Get inputs
+    const vaultName = process.env.INPUT_VAULT;
+    const secretsInput = process.env.INPUT_SECRETS;
+
+    // Parse and normalize secret mappings
+    const secretMappings = secretsInput
+      .split(/[\n,]+/)
+      .map((entry) => entry.trim())
+      .filter((entry) => entry)
+      .map((entry) => {
+        const [input, encoding, output] = entry.split(/(\S+)?>/).map((part) => part?.trim());
+        return { input, encoding, output: output || `\$output:${input}` }; // Default output to $output:input if not specified
+      });
+
+    if (secretMappings.length === 0) {
+      throw new Error('No secrets provided.');
+    }
+
+    // Fetch secrets from Azure Key Vault
+    for (const { input: secretName, encoding, output } of secretMappings) {
+      let secretValue = '';
+
+      const args = encoding
+        ? ['--encoding', encoding, '--file', '/dev/stdout']
+        : ['--query', 'value', '--output', 'tsv'];
+      await exec.exec('az', [
+        'keyvault',
+        'secret',
+        encoding ? 'download' : 'show',
+        '--vault-name',
+        vaultName,
+        '--name',
+        secretName,
+        ...args
+      ], {
+        listeners: {
+          stdout: (data) => {
+            secretValue += data.toString();
+          },
+        },
+      });
+
+      secretValue = secretValue.trim();
+
+      // Mask the secret value in logs
+      core.setSecret(secretValue);
+
+      if (output.startsWith('$env:')) {
+        // Environment variable
+        const envVarName = output.replace('$env:', '').trim();
+        fs.appendFileSync(process.env.GITHUB_ENV, `${envVarName}=${secretValue}\n`);
+        core.info(`Secret set as environment variable: ${envVarName}`);
+      } else if (output.startsWith('$output:')) {
+        // GitHub Actions output variable
+        const outputName = output.replace('$output:', '').trim();
+        core.setOutput(outputName, secretValue.trim());
+        core.info(`Secret set as output variable: ${outputName}`);
+      } else {
+        // File output
+        const filePath = output.trim();
+        const fileDir = filePath.substring(0, filePath.lastIndexOf('/'));
+        if (!fs.existsSync(fileDir)) {
+          fs.mkdirSync(fileDir, { recursive: true });
+        }
```

Comment:
This is a bit fragile in case the `filePath` contains no slash; using [`path.dirname()` ](https://nodejs.org/api/path.html#pathdirnamepath)would be better. And `recursive: true` succeeds if the directory already exists. Maybe something like this?
```suggestion
        fs.mkdirSync(path.dirname(filePath), { recursive: true });
```